### PR TITLE
update docs and add more examples

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: docs/data-sources/azapi_resource.md [hunk 1]
```diff
@@ -64,10 +64,14 @@ output "quarantine_policy" {
 
 The following arguments are supported:
 * `name` - (Required) Specifies the name of the azure resource. Changing this forces a new resource to be created.
-* `parent_id` - (Required) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created.
-  Here're some examples
-  `Container Registry: /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/mygroup1/providers/Microsoft.ContainerRegistry/registries/myregistry1` and
-  `Resource Group: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1`.
+* `parent_id` - (Required) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created. It supports different kinds of deployment scope for **top level** resources: 
+    - resource group scope: `parent_id` should be the ID of a resource group, it's recommended to manage a resource group by `azurerm_resource_group`.
+    - management group scope: `parent_id` should be the ID of a management group, it's recommended to manage a management group by `azurerm_management_group`.
```
Comment (magodo) [inline docs/data-sources/azapi_resource.md:69]: Can we provide the hyperlink to the azurerm registry for these resources?

File: docs/guides/azure_resource_concept.md [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
 ## Scopes:
-Scopes are defined in `bicep-types-az`, however, there're some edge cases like a resource can be deployed in both extension
-and ResourceGroup scope, or a resource's scope is unknown.
+Every resource ID begins with a root scope. There are a few different types of root scopes:
 1. Tenant Scope: id starts with `/`, then followed with `/providers/Microsoft.Foo/bar/{name}`
 2. Subscription Scope: id starts with `/subscriptions/{subscriptionId}`, then followed with `/providers/Microsoft.Foo/bar/{name}`
 3. ResourceGroup Scope: id starts with `/subscriptions/{subscriptionId}/resourceGroups/{groupName}`, then followed with `/providers/Microsoft.Foo/bar/{name}`
-4. Extension Scope: id starts with `{resourceId}`, then followed with `/providers/Microsoft.Foo/bar/{name}`
+4. Extension Scope: id starts with `{resourceID}`, then followed with `/providers/Microsoft.Foo/bar/{name}`
 5. ManagementGroup Scope: id starts with `/providers/Microsoft.Management/managementGroups/{managementGroupName}/`, then followed with `/providers/Microsoft.Foo/bar/{name}`
```
Comment (magodo) [inline docs/guides/azure_resource_concept.md:20]: IIRC, there are also cases that a scope is ended with a location, shall we also mention that here (though it is not mentioned in the ARM wiki)?

File: docs/guides/frequently_asked_questions.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Similarly, values in incorrect casing might be returned and cause a diff. Please use `ignore_casing = true` to suppress it.
+
+## How to manage resources which can be managed by both parent resource API and its own API like `Microsoft.Network/virtualNetworks/subnets`?
+
+For example, `Microsoft.Network/virtualNetworks/subnets` has its own API and it can also be managed by `Microsoft.Network/virtualNetworks` API.
+
+It's recommendded to manage `Microsoft.Network/virtualNetworks/subnets` in `Microsoft.Network/virtualNetworks` instead of using its own API.
+
+The reason is, when update the vnet, its request body won't contain any subnets definitions, so existing subnets will be removed.
```
Comment (magodo) [inline docs/guides/frequently_asked_questions.md:44]: Why is the request body doesn't contain any subnet definitions here?

File: docs/resources/azapi_resource.md [hunk 4]
```diff
@@ -82,10 +82,14 @@ output "quarantine_policy" {
 
 The following arguments are supported:
 * `name` - (Required) Specifies the name of the azure resource. Changing this forces a new resource to be created. 
-* `parent_id` - (Required) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created.
-  Here're some examples 
-  `Container Registry: /subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/mygroup1/providers/Microsoft.ContainerRegistry/registries/myregistry1` and 
-  `Resource Group: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1`.
+* `parent_id` - (Required) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created. It supports different kinds of deployment scope for **top level** resources: 
+    - resource group scope: `parent_id` should be the ID of a resource group, it's recommended to manage a resource group by `azurerm_resource_group`.
+    - management group scope: `parent_id` should be the ID of a management group, it's recommended to manage a management group by `azurerm_management_group`.
```
Comment (magodo) [inline docs/resources/azapi_resource.md:87]: Same here

File: docs/resources/azapi_update_resource.md [hunk 5]
```diff
@@ -93,13 +93,16 @@ resource "azapi_update_resource" "example" {
 The following arguments are supported:
 * `name` - (Optional) Specifies the name of the azure resource. Changing this forces a new resource to be created.
 
-* `parent_id` - (Optional) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created.
+* `parent_id` - (Optional) The ID of the azure resource in which this resource is created. Changing this forces a new resource to be created. It supports different kinds of deployment scope for **top level** resources: 
+    - resource group scope: `parent_id` should be the ID of a resource group, it's recommended to manage a resource group by `azurerm_resource_group`.
+    - management group scope: `parent_id` should be the ID of a management group, it's recommended to manage a management group by `azurerm_management_group`.
```
Comment (magodo) [inline docs/resources/azapi_update_resource.md:98]: Same here

**Review by magodo — COMMENTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: docs/guides/frequently_asked_questions.md (hunk 3 above)
Comment (ms-henglu) [reply docs/guides/frequently_asked_questions.md:44]: I'll add more details to it, the reason is if subnet is defined separately, the vnet contains no definition of subnets, so updating vnet will remove existing subnets.

**Comment by ms-henglu** [comment]
@magodo , thanks for the great suggestions, I've updated the PR, please take another look, thanks!

**Review by magodo — APPROVED** [review]

**Pull request merged**